File: ccif/src/java/other/org/himalay/ccif/NewFlowRequest.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:55 EST 2016")

public  class NewFlowRequest extends   APDUBaseFactory.APDUBase { //Concrete type is NewFlowRequest




    // members variables
    // header
    public APDUHeader header ;
    // length
    public short length ;
    // serviceType
    public short serviceType ;
    // mpegSection
    public ServiceMpegPid mpegSection ;
    // ipUnicast
    public ServiceIPUnicast ipUnicast ;
    // ipMulticast
    public ServiceIPMulticast ipMulticast ;
    // socket
    public ServiceSocket socket ;

    public NewFlowRequest () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize header
        header = new APDUHeader();
        // Initialize length

        // Initialize serviceType

        // Initialize mpegSection
        //Conditional, will be initialized during read
        // Initialize ipUnicast
        //Conditional, will be initialized during read
        // Initialize ipMulticast
        //Conditional, will be initialized during read
        // Initialize socket
        //Conditional, will be initialized during read
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        preRead();
        int retVal= 0;
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read serviceType
        {
            serviceType=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read mpegSection
        if (serviceType==0x00) {
            mpegSection = new ServiceMpegPid();
            retVal += mpegSection.read(istream);
        }
        // read ipUnicast
        if (serviceType==0x01) {
            ipUnicast = new ServiceIPUnicast();
            retVal += ipUnicast.read(istream);
        }
        // read ipMulticast
        if (serviceType==0x02) {
            ipMulticast = new ServiceIPMulticast();
            retVal += ipMulticast.read(istream);
        }
        // read socket
        if (serviceType==0x04) {
            socket = new ServiceSocket();
            retVal += socket.read(istream);
        }

        postRead();
        return retVal;
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read header
        retVal += header.read(istream);
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read serviceType
        {
            serviceType=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read mpegSection
        if (serviceType==0x00) {
            mpegSection = new ServiceMpegPid();
            retVal += mpegSection.read(istream);
        }
        // read ipUnicast
        if (serviceType==0x01) {
            ipUnicast = new ServiceIPUnicast();
            retVal += ipUnicast.read(istream);
        }
        // read ipMulticast
        if (serviceType==0x02) {
            ipMulticast = new ServiceIPMulticast();
            retVal += ipMulticast.read(istream);
        }
        // read socket
        if (serviceType==0x04) {
            socket = new ServiceSocket();
            retVal += socket.read(istream);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;

        {   /** fix dependent sizes for header **/
        }


        {   /** fix dependent sizes for mpegSection **/
        }
        {   /** fix dependent sizes for ipUnicast **/
        }
        {   /** fix dependent sizes for ipMulticast **/
        }
        {   /** fix dependent sizes for socket **/
        }

        // write header
        if (header!=null)retVal +=header.write(ostream);
        // write length
        ostream.writeByte(length);
        retVal +=1;
        // write serviceType
        ostream.writeByte(serviceType);
        retVal +=1;
        // write mpegSection
        if (mpegSection!=null)retVal +=mpegSection.write(ostream);
        // write ipUnicast
        if (ipUnicast!=null)retVal +=ipUnicast.write(ostream);
        // write ipMulticast
        if (ipMulticast!=null)retVal +=ipMulticast.write(ostream);
        // write socket
        if (socket!=null)retVal +=socket.write(ostream);
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("NewFlowRequest\n");
        dc.increaseIndent();
        int retVal= 0;
        // write header
        if ( header != null ) {
            dc.indent();
            dc.getPs().println("header") ;
            retVal +=header.dump(dc);
        }
        // write length
        dc.indent();
        dc.getPs().println("length="+length+"(0x"+ Integer.toHexString(length) + ")" );
        // write serviceType
        dc.indent();
        dc.getPs().println("serviceType="+serviceType+"(0x"+ Integer.toHexString(serviceType) + ")" );
        // write mpegSection
        if ( mpegSection != null ) {
            dc.indent();
            dc.getPs().println("mpegSection") ;
            retVal +=mpegSection.dump(dc);
        }
        // write ipUnicast
        if ( ipUnicast != null ) {
            dc.indent();
            dc.getPs().println("ipUnicast") ;
            retVal +=ipUnicast.dump(dc);
        }
        // write ipMulticast
        if ( ipMulticast != null ) {
            dc.indent();
            dc.getPs().println("ipMulticast") ;
            retVal +=ipMulticast.dump(dc);
        }
        // write socket
        if ( socket != null ) {
            dc.indent();
            dc.getPs().println("socket") ;
            retVal +=socket.dump(dc);
        }
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for header
    //public APDUHeader getHeader()
    //{
    //    return header ;
    //}


    // Setter for header
    //public void setHeader(APDUHeader val)
    //{
    //    this.header= val;
    //}
    // Getter for length
    //public short getLength()
    //{
    //    return length ;
    //}


    // Setter for length
    //public void setLength(short val)
    //{
    //    this.length= val;
    //}
    // Getter for serviceType
    //public short getServiceType()
    //{
    //    return serviceType ;
    //}


    // Setter for serviceType
    //public void setServiceType(short val)
    //{
    //    this.serviceType= val;
    //}
    // Getter for mpegSection
    //public ServiceMpegPid getMpegSection()
    //{
    //    return mpegSection ;
    //}


    // Setter for mpegSection
    //public void setMpegSection(ServiceMpegPid val)
    //{
    //    this.mpegSection= val;
    //}
    // Getter for ipUnicast
    //public ServiceIPUnicast getIpUnicast()
    //{
    //    return ipUnicast ;
    //}


    // Setter for ipUnicast
    //public void setIpUnicast(ServiceIPUnicast val)
    //{
    //    this.ipUnicast= val;
    //}
    // Getter for ipMulticast
    //public ServiceIPMulticast getIpMulticast()
    //{
    //    return ipMulticast ;
    //}


    // Setter for ipMulticast
    //public void setIpMulticast(ServiceIPMulticast val)
    //{
    //    this.ipMulticast= val;
    //}
    // Getter for socket
    //public ServiceSocket getSocket()
    //{
    //    return socket ;
    //}


    // Setter for socket
    //public void setSocket(ServiceSocket val)
    //{
    //    this.socket= val;
    //}


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }


    public void setHeader(APDUHeader header)
    {
        this.header= header;
    }

    public APDUHeader getHeader()
    {
        return this.header;
    }

}

// End of code

File: libccif/src/bp/NewFlowRequest.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:12 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "NewFlowRequest.h"

#ifdef org_himalay_ccif_NewFlowRequest__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
NewFlowRequest::NewFlowRequest() {
    references= 0;


    // header
    org_himalay_ccif_APDUHeader__NEW( header);
    // length
    length= (ui8)0;
    // serviceType
    serviceType= (ui8)0;
    // mpegSection
    org_himalay_ccif_ServiceMpegPid__INIT(mpegSection)
    // ipUnicast
    org_himalay_ccif_ServiceIPUnicast__INIT(ipUnicast)
    // ipMulticast
    org_himalay_ccif_ServiceIPMulticast__INIT(ipMulticast)
    // socket
    org_himalay_ccif_ServiceSocket__INIT(socket)
}
// Destructor
NewFlowRequest::~NewFlowRequest() {
    // header
    org_himalay_ccif_APDUHeader__DELETE( header);
    // length

    // serviceType

    // mpegSection
    org_himalay_ccif_ServiceMpegPid__DELETE( mpegSection);
    // ipUnicast
    org_himalay_ccif_ServiceIPUnicast__DELETE( ipUnicast);
    // ipMulticast
    org_himalay_ccif_ServiceIPMulticast__DELETE( ipMulticast);
    // socket
    org_himalay_ccif_ServiceSocket__DELETE( socket);
}

int NewFlowRequest::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // serviceType
    {
        serviceType=(istream.readUI8());
        retVal+=1;
    }
    // mpegSection
    if (serviceType==0x00) {
        org_himalay_ccif_ServiceMpegPid__NEW( mpegSection);
        retVal += mpegSection->read(istream);
    }
    // ipUnicast
    if (serviceType==0x01) {
        org_himalay_ccif_ServiceIPUnicast__NEW( ipUnicast);
        retVal += ipUnicast->read(istream);
    }
    // ipMulticast
    if (serviceType==0x02) {
        org_himalay_ccif_ServiceIPMulticast__NEW( ipMulticast);
        retVal += ipMulticast->read(istream);
    }
    // socket
    if (serviceType==0x04) {
        org_himalay_ccif_ServiceSocket__NEW( socket);
        retVal += socket->read(istream);
    }
    return retVal;
}

int NewFlowRequest::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read header
    retVal += header->read(istream);
    // read length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // read serviceType
    {
        serviceType=(istream.readUI8());
        retVal+=1;
    }
    // read mpegSection
    if (serviceType==0x00) {
        org_himalay_ccif_ServiceMpegPid__NEW( mpegSection);
        retVal += mpegSection->read(istream);
    }
    // read ipUnicast
    if (serviceType==0x01) {
        org_himalay_ccif_ServiceIPUnicast__NEW( ipUnicast);
        retVal += ipUnicast->read(istream);
    }
    // read ipMulticast
    if (serviceType==0x02) {
        org_himalay_ccif_ServiceIPMulticast__NEW( ipMulticast);
        retVal += ipMulticast->read(istream);
    }
    // read socket
    if (serviceType==0x04) {
        org_himalay_ccif_ServiceSocket__NEW( socket);
        retVal += socket->read(istream);
    }

    return retVal;
}

int NewFlowRequest::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for header **/
    }


    {   /** fix dependent sizes for mpegSection **/
    }
    {   /** fix dependent sizes for ipUnicast **/
    }
    {   /** fix dependent sizes for ipMulticast **/
    }
    {   /** fix dependent sizes for socket **/
    }

    int retVal= 0;
    // write header
    if ( header != NULL ) {
        retVal +=header->write(ostream);
    }
    // write length
    ostream.writeUI8(length);
    retVal +=1;
    // write serviceType
    ostream.writeUI8(serviceType);
    retVal +=1;
    // write mpegSection
    if ( mpegSection != NULL ) {
        retVal +=mpegSection->write(ostream);
    }
    // write ipUnicast
    if ( ipUnicast != NULL ) {
        retVal +=ipUnicast->write(ostream);
    }
    // write ipMulticast
    if ( ipMulticast != NULL ) {
        retVal +=ipMulticast->write(ostream);
    }
    // write socket
    if ( socket != NULL ) {
        retVal +=socket->write(ostream);
    }

    return retVal;
}

int NewFlowRequest::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("NewFlowRequest\n");
    dc.increaseIndent();
    int retVal= 0;
    // write header
    if ( header != NULL ) {
        dc.indent();
        dc.getPs().println("header") ;
        retVal +=header->dump(dc);
    }
    // write length
    dc.indent();
    dc.getPs().print("length=");
    dc.getPs().println((long)length);
    // write serviceType
    dc.indent();
    dc.getPs().print("serviceType=");
    dc.getPs().println((long)serviceType);
    // write mpegSection
    if ( mpegSection != NULL ) {
        dc.indent();
        dc.getPs().println("mpegSection") ;
        retVal +=mpegSection->dump(dc);
    }
    // write ipUnicast
    if ( ipUnicast != NULL ) {
        dc.indent();
        dc.getPs().println("ipUnicast") ;
        retVal +=ipUnicast->dump(dc);
    }
    // write ipMulticast
    if ( ipMulticast != NULL ) {
        dc.indent();
        dc.getPs().println("ipMulticast") ;
        retVal +=ipMulticast->dump(dc);
    }
    // write socket
    if ( socket != NULL ) {
        dc.indent();
        dc.getPs().println("socket") ;
        retVal +=socket->dump(dc);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for header
//org_himalay_ccif_APDUHeader__PTR_TYPE NewFlowRequest::getHeader()
//{
//return this->header;
//}

// Setter for header
//void NewFlowRequest::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE val)
//{
//this->header= val;
//}

// Getter for length
//ui8 NewFlowRequest::getLength()
//{
//return this->length;
//}

// Setter for length
//void NewFlowRequest::setLength(ui8 val)
//{
//this->length= val;
//}

// Getter for serviceType
//ui8 NewFlowRequest::getServiceType()
//{
//return this->serviceType;
//}

// Setter for serviceType
//void NewFlowRequest::setServiceType(ui8 val)
//{
//this->serviceType= val;
//}

// Getter for mpegSection
//org_himalay_ccif_ServiceMpegPid__PTR_TYPE NewFlowRequest::getMpegSection()
//{
//return this->mpegSection;
//}

// Setter for mpegSection
//void NewFlowRequest::setMpegSection(org_himalay_ccif_ServiceMpegPid__PTR_TYPE val)
//{
//this->mpegSection= val;
//}

// Getter for ipUnicast
//org_himalay_ccif_ServiceIPUnicast__PTR_TYPE NewFlowRequest::getIpUnicast()
//{
//return this->ipUnicast;
//}

// Setter for ipUnicast
//void NewFlowRequest::setIpUnicast(org_himalay_ccif_ServiceIPUnicast__PTR_TYPE val)
//{
//this->ipUnicast= val;
//}

// Getter for ipMulticast
//org_himalay_ccif_ServiceIPMulticast__PTR_TYPE NewFlowRequest::getIpMulticast()
//{
//return this->ipMulticast;
//}

// Setter for ipMulticast
//void NewFlowRequest::setIpMulticast(org_himalay_ccif_ServiceIPMulticast__PTR_TYPE val)
//{
//this->ipMulticast= val;
//}

// Getter for socket
//org_himalay_ccif_ServiceSocket__PTR_TYPE NewFlowRequest::getSocket()
//{
//return this->socket;
//}

// Setter for socket
//void NewFlowRequest::setSocket(org_himalay_ccif_ServiceSocket__PTR_TYPE val)
//{
//this->socket= val;
//}

int NewFlowRequest::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

void NewFlowRequest::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE header)
{
    this->header= header;
}

org_himalay_ccif_APDUHeader__PTR_TYPE NewFlowRequest::getHeader()
{
    return this->header;
}

#ifdef org_himalay_ccif_NewFlowRequest__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::NewFlowRequest* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::NewFlowRequest* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code